### PR TITLE
Optimize Kernel.in/2 for empty lists

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/lib/kernel.ex [hunk 1]
```diff
@@ -2676,10 +2676,10 @@ defmodule Kernel do
     end
 
     case right do
-      _ when in_module? ->
```
Comment (lexmag) [inline lib/elixir/lib/kernel.ex]: I think the order is important here as we could be interested in possible side-effects in `unquote(left)`.

Comment (eksperimental) [inline lib/elixir/lib/kernel.ex]: @lexmag do you have an example that could represent what you are saying? thanks

Comment (lexmag) [inline lib/elixir/lib/kernel.ex]: For instance:
```elixir
db_insertion() in []
```
with new order will be rewritten as simple `false` and `db_insertion()` will never happen.

Comment (eksperimental) [inline lib/elixir/lib/kernel.ex]: excellent! all clear now

**Pull request closed without merging**